Lib/test/test_capi/test_file.py

```diff
@@ -0,0 +1,23 @@
+import os
+import unittest
+from test.support import import_helper
+
+_testcapi = import_helper.import_module('_testcapi')
+
+
+class CAPIFileTest(unittest.TestCase):
+    def test_py_fopen(self):
```

Comment:
TODO: Test with TESTFN, TESTFN_UNENCODABLE, TESTFN_UNDECODABLE and corresponding encoded/decoded variants. Test with NULL, embedded null bytes/characters.